google-cloud-storage/src/test/resources/com/google/cloud/storage/conformance/retry/testNamesWhichShouldSucceed.txt

```diff
@@ -0,0 +1,322 @@
+# Each line should be a full test name
+# Each test name present in the file will be expected to pass in a CI environment
+#
+# This list can be regenerated by running the following command in a shell:
+#    mvn test && xq '//testsuite/testcase[not(./error) and not(./failure) and not(./skipped)]/@name' google-cloud-storage/target/surefire-reports/TEST-com.google.cloud.storage.conformance.retry.ITRetryConformanceTest-sponge_log.xml | tail -n+2 | head -n-1 | sed 's#  <result>test\[##g' | sed 's#\]</result>##g' | sort | tee -a google-cloud-storage/src/test/resources/com/google/cloud/storage/conformance/retry/testNamesWhichShouldSucceed.txt
+# where xq is the package from https://github.com/jeffbr13/xq
+
+TestRetryConformance/1-[return-503_return-503]-storage.bucket_acl.list-7
+TestRetryConformance/1-[return-503_return-503]-storage.bucket_acl.list-8
+TestRetryConformance/1-[return-503_return-503]-storage.bucket_acl.list-90
```

Comment:
nit: Do we have to follow the naming guidelines from the design doc? Since there are way less instruction cases than mapping library methods, perhaps it would make more sense to expose the "library method name" and use "instruction index" instead?

Reply:
in the case of java, there are often multiple library method names which are underlying these. 

In this case for bucket_acl.list
* ` 7` -> `com.google.cloud.storage.Storage.listAcls(String bucketName)`
* ` 8` -> `com.google.cloud.storage.Storage.listAcls(String bucketName, BucketSourceOption... options)`
* `90` -> `com.google.cloud.storage.Bucket.listAcls()`

When looking at the results of a test run, I think it's easier for someone to read the lines as defined by the design doc with the understanding, there are several language specific under each of the final index but knowing which instructions were used without having to memorize or refer to indexes from the test definition.

Use the instruction names are also resilient to any re-ordering which may happen as we add more cases in the future.